setup.py

```diff
@@ -101,9 +101,33 @@ def get_config_schema():
 def main():
     import sys
 
+    from setuptools import Extension
+    from setuptools.command.build_ext import build_ext
     from aksetup_helper import (hack_distutils, get_config, setup,
-            NumpyExtension, set_up_shipped_boost_if_requested,
-            check_git_submodules)
+            set_up_shipped_boost_if_requested, check_git_submodules)
+
+    class NumpyExtension(Extension):
+        pass
+
+    class custom_build_ext(build_ext):
+
+        def build_extension(self, extension):
+            # We add the numpy include dir right before building the
+            # extension, in order to avoid having to import numpy when
+            # the setup script is imported, which would prevent
+            # installation before manual installation of numpy.
+            if isinstance(extension, NumpyExtension):
+                numpy_incpath = self.get_numpy_incpath()
+                if numpy_incpath not in extension.include_dirs:
+                    extension.include_dirs.append(numpy_incpath)
+            super().build_extension(extension)
+
+        def get_numpy_incpath(self):
+            from imp import find_module
+            # avoid actually importing numpy, it screws up distutils
+            file, pathname, descr = find_module("numpy")
+            from os.path import join
+            return join(pathname, "core", "include")
```

Comment:
Minor note: `get_numpy_incpath` doesn't access any instance variables and can therefore be defined outside the `custom_build_ext` class.

Reply:
Ah true. I could even have reused the `NumpyExtension.get_numpy_incpath` in `aksetup_helper.py` from which it was copied (through would have to make an instance of its `NumpyExtension`). I'll wait and see what @inducer says, he may even want me to patch `aksetup_helper.py` with this stuff instead.